### PR TITLE
FF123 Nightly - Web Codecs API video support on linux

### PR DESCRIPTION
**Pull request by hamishwillee**
FF123 supports the "video" parts of the Web Codecs API on nightly builds (pref  `pref dom.media.webcodecs.enabled`) in https://bugzilla.mozilla.org/show_bug.cgi?id=1749047 - but at time of writing only on Linux.

According to the developer full support is intended for the supported methods (not just Linux). This updates according to add support in preview, using partial implementation and a note to indicate only Linux support. 

Confirmation from developer here: https://bugzilla.mozilla.org/show_bug.cgi?id=1749047#c80

Related docs work in https://github.com/mdn/content/issues/31913

### CODE REVIEW

File: api/EncodedVideoChunk.json [hunk 1]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (queengooborg) [inline api/EncodedVideoChunk.json:19]:
```suggestion
            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/EncodedVideoChunk.json [hunk 2]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:19]:
```suggestion
            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```

File: api/EncodedVideoChunk.json [hunk 3]
```diff
@@ -88,7 +92,9 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview",
+              "partial_implementation": true,
+              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:97]:
```suggestion
              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```

File: api/EncodedVideoChunk.json [hunk 4]
```diff
@@ -125,7 +131,9 @@
...
```
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:136]:
```suggestion
              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```

File: api/EncodedVideoChunk.json [hunk 5]
```diff
@@ -51,7 +53,9 @@
...
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply api/EncodedVideoChunk.json:58]: No worries - I clearly just thought "that sounds OK and did do further checking".

```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/EncodedVideoChunk.json (hunk 3 above)
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:97]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/EncodedVideoChunk.json (hunk 4 above)
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:136]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/EncodedVideoChunk.json [hunk 6]
```diff
@@ -199,7 +207,9 @@
...
```
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:212]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/EncodedVideoChunk.json [hunk 7]
```diff
@@ -236,7 +246,9 @@
...
```
Comment (hamishwillee) [inline api/EncodedVideoChunk.json:251]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 8]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:19]:
```suggestion
            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 9]
```diff
@@ -51,7 +53,9 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview",
+              "partial_implementation": true,
+              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:58]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 10]
```diff
@@ -88,7 +92,9 @@
...
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:97]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 11]
```diff
@@ -162,7 +170,9 @@
...
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:175]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 12]
```diff
@@ -199,7 +209,9 @@
...
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:214]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoColorSpace.json [hunk 13]
```diff
@@ -236,7 +248,9 @@
...
```
Comment (hamishwillee) [inline api/VideoColorSpace.json:253]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 14]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoDecoder.json:19]:
```suggestion
            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 15]
```diff
@@ -51,7 +53,9 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview",
+              "partial_implementation": true,
+              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoDecoder.json:58]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 16]
```diff
@@ -88,7 +92,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:97]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 17]
```diff
@@ -125,7 +131,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:136]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 18]
```diff
@@ -162,7 +170,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:175]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 19]
```diff
@@ -199,7 +209,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:214]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 20]
```diff
@@ -236,7 +248,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:253]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 21]
```diff
@@ -273,7 +287,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:292]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 22]
```diff
@@ -310,7 +326,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:331]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 23]
```diff
@@ -347,7 +365,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:370]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoDecoder.json [hunk 24]
```diff
@@ -384,7 +404,9 @@
...
```
Comment (hamishwillee) [inline api/VideoDecoder.json:409]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 25]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoEncoder.json:19]:
```suggestion
            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 26]
```diff
@@ -51,7 +53,9 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview",
+              "partial_implementation": true,
+              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoEncoder.json:58]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 27]
```diff
@@ -88,7 +92,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:97]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 28]
```diff
@@ -125,7 +131,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:136]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 29]
```diff
@@ -162,7 +170,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:175]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 30]
```diff
@@ -199,7 +209,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:214]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 31]
```diff
@@ -236,7 +248,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:253]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 32]
```diff
@@ -309,7 +325,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:330]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 33]
```diff
@@ -346,7 +364,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:369]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoEncoder.json [hunk 34]
```diff
@@ -383,7 +403,9 @@
...
```
Comment (hamishwillee) [inline api/VideoEncoder.json:408]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 35]
```diff
@@ -14,7 +14,9 @@
           "chrome_android": "mirror",
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "preview",
+            "partial_implementation": true,
+            "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoFrame.json:19]:
```suggestion
            "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 36]
```diff
@@ -51,7 +53,9 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview",
+              "partial_implementation": true,
+              "notes": "Supported on Linux desktop only (<a href='https://bugzil.la/1749047'>bug 1749047</a>)."
```
Comment (hamishwillee) [inline api/VideoFrame.json:58]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 37]
```diff
@@ -88,7 +92,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:97]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 38]
```diff
@@ -125,7 +131,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:136]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 39]
```diff
@@ -162,7 +170,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:175]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 40]
```diff
@@ -199,7 +209,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:214]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 41]
```diff
@@ -236,7 +248,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:253]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 42]
```diff
@@ -273,7 +287,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:292]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 43]
```diff
@@ -310,7 +326,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:331]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 44]
```diff
@@ -346,7 +364,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:369]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 45]
```diff
@@ -420,7 +442,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:447]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 46]
```diff
@@ -457,7 +481,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:486]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 47]
```diff
@@ -494,7 +520,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:525]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 48]
```diff
@@ -531,7 +559,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:564]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

File: api/VideoFrame.json [hunk 49]
```diff
@@ -568,7 +598,9 @@
...
```
Comment (hamishwillee) [inline api/VideoFrame.json:603]:
```suggestion
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

**Comment by hamishwillee** [comment]
This should be ready for re-review @queengooborg - I went through and fixed all problem cases with your new note.

File: api/EncodedVideoChunk.json [hunk 50]
```diff
@@ -162,7 +170,7 @@
             "chrome_android": "mirror",
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "preview"
```
Comment (queengooborg) [inline api/EncodedVideoChunk.json:173]: Should this also have the partial implementation and note?

Comment (hamishwillee) [reply api/EncodedVideoChunk.json:173]: Sure :-)
```suggestion
              "version_added": "preview",
              "partial_implementation": true,
              "notes": "Only supported on Linux, see <a href='https://bugzil.la/1749047'>bug 1749047</a>."
```

**Comment by hamishwillee** [comment]
Here we go again @queengooborg :-)

**Review by foolip — APPROVED** [review]
Everything looks very uniform, nothing much to comment on here :)

**Pull request merged**